kDriveCore/Data/Cache/DriveFileManager/DriveFileManager.swift

```diff
@@ -1018,22 +1018,30 @@ public final class DriveFileManager {
     }
 
     func removeFileInDatabase(fileUid: String, cascade: Bool, writableRealm: Realm) {
-        guard let liveFile = writableRealm.object(ofType: File.self, forPrimaryKey: fileUid), !liveFile.isInvalidated else {
-            return
-        }
+        var fileUidsToProcess = [fileUid]
+        var liveFilesToDelete = [File]()
 
-        let localContainerUrl = liveFile.localContainerUrl
-        if fileManager.fileExists(atPath: localContainerUrl.path) {
-            try? fileManager.removeItem(at: localContainerUrl) // Check that it was correctly removed?
-        }
+        while !fileUidsToProcess.isEmpty {
+            let currentFileUid = fileUidsToProcess.removeLast()
 
-        if cascade {
-            for frozenChild in liveFile.children.freeze() where !frozenChild.isInvalidated {
-                removeFileInDatabase(fileUid: frozenChild.uid, cascade: cascade, writableRealm: writableRealm)
+            if let file = writableRealm.object(ofType: File.self, forPrimaryKey: currentFileUid), !file.isInvalidated {
+                if fileManager.fileExists(atPath: file.localContainerUrl.path) {
+                    try? fileManager.removeItem(at: file.localContainerUrl)
+                }
+
+                if cascade {
+                    let filesUidsToDelete = liveFilesToDelete.map { $0.uid }
```

Comment:
Had to track `uid` in a dedicated object. I was getting `modification-requires-exclusive-access` crash if checking against `liveFilesToDelete` directly